README.adoc

```diff
@@ -1511,6 +1511,110 @@ assertThat(new File("target/test-classes/index.epub").exists(), is(true));
+- Implement the converter as a subclass of `org.asciidoctor.converter.AbstractConverter`.
+- Register the converter with a well defined backend name at the `ConverterRegistry`.
+- Pass the backend name to the `Asciidoctor` instance when rendering a source file.
+
+A basic converter that converts to an own text format looks like this:
+
+[source,java]
+.org.asciidoctor.converter.TextConverter.java
+----
+public class TextConverter extends AbstractConverter {
+
+    private String LINE_SEPARATOR = "\n";
+
+    public TextConverter(String backend, Map<Object, Object> opts) {         // <1>
+        super(backend, opts);
+    }
+    
+    @Override
+    public Object convert(
+            AbstractNode node, String transform, Map<Object, Object> o) {    // <2>
+
+        if (transform == null) {                                             // <3>
+            transform = node.getNodeName();
+        }
+ 
+        if (node instanceof DocumentRuby) {
+            DocumentRuby document = (DocumentRuby) node;
+            return document.getContent();                                    // <4>
+        } else if (node instanceof Section) {
+            Section section = (Section) node;
+            return new StringBuilder()
+                    .append("== ").append(section.getTitle()).append(" ==")
+                    .append(LINE_SEPARATOR).append(LINE_SEPARATOR)
+                    .append(section.getContent()).toString();                // <4>
+        } else if (transform.equals("paragraph")) {
+            AbstractBlock block = (AbstractBlock) node;
+            String content = (String) block.content();                       // <4>
+            return new StringBuilder(content.replaceAll(LINE_SEPARATOR, " "))
+                            .append(LINE_SEPARATOR);
+        } 
+        return null;
+    }
+
+}
+----
+<1> A converter must implement this constructor, because {uri-repo}[AsciidoctorJ] will call the constructor with this signature.
+    For every conversion a new instance will be created.
+<2> The method `convert()` is called with the AST object for the document, i.e. a `DocumentRuby` instance, when a document is rendered.
+<3> The optional parameter `transform` hints at the transformation to be executed.
+    This could be for example the value `embedded` to indicate that the resulting document should be without headers and footers.
+    If it is `null` the transformation usually is defined by the node type and name.
+<4> Calls to the method `getContent()` of a node will recursively call the method `convert()` with the child nodes again.
+    Thereby the converter can collect the rendered child nodes, merge them appropriately and return the rendering of the whole node.
+
+Then the converter has to be registered like this with a well defined backend name:
+
+[source,java]
+.Register a converter
+----
+Asciidoctor asciidoctor = Asciidoctor.Factory.create();
+asciidoctor.javaConverterRegistry().register(TextConverter.class, "text"); // <1>
```

Comment:
I wonder if it would be possible to read the backend name from the converter class so that it is optional when registering it. Something like:

```
register(TextConverter.class);
```